.github/workflows/t-bytecode-compare.yml

```diff
@@ -29,6 +29,30 @@ jobs:
     steps:
       - uses: actions/setup-node@v4
 
+      - name: Free Disk Space Before Build
+        if: "env.PLATFORM == 'macosx-amd64'"
+        run: |
+          df -hI /dev/disk3s1s1
+          sudo rm -rf /Applications/Xcode_14.3.1.app
+          sudo rm -rf /Applications/Xcode_15.0.1.app
+          sudo rm -rf /Applications/Xcode_15.1.app
+          sudo rm -rf /Applications/Xcode_15.2.app
+          sudo rm -rf /Applications/Xcode_15.3.app
```

Comment:
Is this always installed? I'm wondering if this will not break eventually. I.e. when the base `macos-latest` image stop to provide them (currently it is macos-14): https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md

Reply:
Maybe we need to create an issue to rethink this whole cleanup process before the next release.

Reply:
Oh, you used `--force`, so it will ignore nonexistent files. So yeah, it will not break.

Reply:
IMO this shouldn't be more than a temporary fix in any case.

Reply:
> So yeah, it will not break.

Well, it *will* break, because when it stops deleting stuff, we'll run out of space again :)

Not sure if we can do better though. The macOS runner does not have the extra data partition so to fit the repo in we do have to delete stuff. We should just make this a bit more robust by not depending on specific versions of stuff we delete.

Reply:
An alternative would be to go back to deleting old nightlies. We have so many now that [github crashes when you go to `bin/`](https://github.com/ethereum/solc-bin/tree/gh-pages/bin). It's just that it also does not feel like a proper solution because we can't delete them from history so we're still storing them there. The repo will still keep growing in an unbounded way, only the checkout will be smaller.

I don't think it was ever a good idea to store nightly binaries in a github repo (they really should have been stored as something less permanent, e.g. pre-releases, if at all).

Reply:
You can access them e.g. in Remix, so some people may be using them. Some frameworks may be making them available to users too - I remember e.g. that there was some interest from Truffle in having them accessible in the new directory structure. But I don't think anyone is relying on them in production. And they do, we don't have to support that (I wonder if e.g. Sourcify does?).

I added this for discussion next Wednesday. Perhaps it's time to drop some historical baggage and simplify things. I'd rather avoid half-measures, but I'm also not sure how far we can safely go.

TBH I wouldn't be against just scrapping those nightlies completely. They're a nuisance while at the same time they are not useful enough. It's often much more convenient to get a platform-specific binary from `develop` and we don't have those here. And also it's often more useful for us to share a build from an unmerged PR. Both of these things we currently get from CircleCI and we don't even notice when the nightlies here break. CircleCI only stores artifacts for 30 days, but that's usually more than enough. The only annoying bit is that giving them to people always requires explanations, because links are a buried deep in the CCI UI.